Add a issue template for tracking minor releases

kennyadsl opened the pull request:
## Summary

When it's time to release a new minor version, we can use this template to create a new issue that encapsules all the steps needed for complete the release.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

rainerdema commented on .github/ISSUE_TEMPLATE/new_minor_release.md:
```diff
@@ -0,0 +1,67 @@
+### Steps to do outside the main repository
+
+- [ ] [[Guides](https://github.com/solidusio/edgeguides/)] Publish upgrade instruction for the new version ([e.g. PR](https://github.com/solidusio/edgeguides/pull/78)).
+- [ ] [[Starter Frontend](https://github.com/solidusio/solidus_starter_frontend)] Create a new matching branch (e.g. v4.1)
+- [ ] [[Starter Frontend](https://github.com/solidusio/solidus_starter_frontend)] In the new branch (e.g. `v3.3`), change the [default solidus target branch in the CircleCI Config](https://github.com/solidusio/solidus_starter_frontend/blob/4fd99d229cefbe22e7a3916a6dcb6a0cc6cbd41b/.circleci/config.yml#L96) to point to the newly created branch.
```
These two steps use different versions as examples (v4.1 and v3.3), even if I have to change the default solidus target branch in the CircleCI Config in the previously created branch.
If so, this might confuse the readers.

kennyadsl replied:
Thanks for spotting that @rainerdema, fixing!

waiting-for-dev commented on .github/ISSUE_TEMPLATE/new_minor_release.md:
```diff
@@ -0,0 +1,67 @@
+- Be sure there are no pending PRs that you want to include in the next minor.
+- Visit the GitHub Solidus repository.
+- Navigate to the "Actions" tab and select the "Prepare release" item.
+- Click the "Run Workflow" button, leaving `main` as the branch and the checkbox to prepare the next major unchecked (unless necesasy).
```
```suggestion
- Click the "Run Workflow" button, leaving `main` as the branch and the checkbox to prepare the next major unchecked (unless necessary).
```

waiting-for-dev replied:
Although, if the issue is only used for minor releases, it shouldn't b ever necessary :slightly_smiling_face:

kennyadsl replied:
But the last minor still needs a minor release, isn't it? Or in that case the last minor is covered in the major release procedure?

waiting-for-dev replied:
Oh, yeah, sorry, I got confused :slightly_smiling_face:

waiting-for-dev commented on .github/ISSUE_TEMPLATE/new_minor_release.md:
````diff
@@ -0,0 +1,67 @@
+Open Solidus on your local machine and run:
+
+```bash
+git fetch -a
````
`fetch` won't update the `main` branch, right? We still need to reset the local branch or use `pull` instead.

kennyadsl replied:
I think that if you are on another branch and then checkout main, you should have the latest version from the remote. But even if that's true, what I wrote doesn't cover all the scenarios, you are right, I'll just change it to 

```
git checkout main
git pull
gem release
```

elia replied:
pull can generate merge commits or rebases depending on the configuration, `reset --hard` is a maybe safer alternative:

```
git checkout main
git fetch -a
git reset --hard origin/main
gem release
```

kennyadsl replied:
I wanted to avoid using `origin` because not everyone (me included) has that as the main Solidus repo remote. It's so hard to update a git repo in a way that works for everyone! 🤣

waiting-for-dev reviewed (commented):
I love it! :heart_eyes:

I was wondering if, at this point, we could have the wiki page link to the template so that we don't have to maintain the information in two different places.

kennyadsl commented:
> I was wondering if, at this point, we could have the wiki page link to the template so that we don't have to maintain the information in two different places.

I thought the same. At this point, I can probably do it immediately and make this the only source of truth.

kennyadsl commented:
I reworked the template a little bit, changing:

- Removed references to the "How to release Solidus" wiki page, in order to be ready to invert the link once this is merged.
- Added REST API doc update that I did miss.
- Added docusaurus version update that I did miss.

rainerdema reviewed (approved):
Excellent work! ❤️ 
As soon as I have the chance to progress through this release process, I may find other areas for improvement. But from what I'm reading now, I think it's well-structured and detailed, which will be extremely helpful.

kennyadsl commented on .github/ISSUE_TEMPLATE/new_minor_release.md:
```diff
@@ -0,0 +1,73 @@
+
+- [ ] Post-Release PR Merged!
+
+### Steps to do outside the main repository
```
That's a good point. At the moment all of these steps are done by the same person but once we can split them, it'd definitely a good idea to separate them here as well. 👍
